presto-memory/src/test/java/com/facebook/presto/plugin/memory/TestMemoryDistributedQueries.java

```diff
@@ -0,0 +1,20 @@
+package com.facebook.presto.plugin.memory;
+
+import com.facebook.presto.tests.AbstractTestDistributedQueries;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.Test;
+
+public class TestMemoryDistributedQueries
+        extends AbstractTestDistributedQueries
```

Comment:
There are lots of tests that won't pass for the memory connector. Instead, add a similar test to `TestMemorySmoke`. We can skip the parts that are meant to test the engine (like joins), so something basic like this:
```java
@Language("SQL") String query = "SELECT orderkey, orderstatus, totalprice / 2 half FROM orders";

assertUpdate("CREATE VIEW test_view AS SELECT 123 x");
assertUpdate("CREATE OR REPLACE VIEW test_view AS " + query);

assertQueryFails("CREATE TABLE test_view (x date)", "View already exists: tpch.test_view");
assertQueryFails("CREATE VIEW test_view AS SELECT 123 x", "View already exists: tpch.test_view");

assertQuery("SELECT * FROM test_view", query);

assertTrue(computeActual("SHOW TABLES").getOnlyColumnAsSet().contains("test_view"));

assertUpdate("DROP VIEW test_view");
assertQueryFails("DROP VIEW test_view", "View 'tpch.test_view' not found");
```

Reply:
Oh wow, sorry I thought that if I override just testView(), testng would only run testView. You're right, I'll fix this.